### PR TITLE
Sorts periods chronologically.

### PR DESCRIPTION
**Pull request by YushYush**
Resolves issue #186 
We can sort at zero cost by sorting period times if and only if period data is updated. The API will now guarantee that periods will be returned in sorted order.

### CODE REVIEW

File: app/models/section.rb [hunk 1]
```diff
@@ -20,4 +21,16 @@ def conflicts_with(section)
...
+
+  def sort_periods
+    periods_info = periods_day.zip(periods_start, periods_end, periods_type)
+    periods_info = periods_info.sort!.transpose
+    self.periods_day, self.periods_start, self.periods_end, self.periods_type = periods_info
+  end
+
+  def period_info_changed?
+    if(self.changed & ["periods_start", "periods_end", "periods_day", "periods_type"]).any?
+      return true
```
Comment (Bad-Science) [inline app/models/section.rb]: Control flow is not needed, and `return` should be omitted. This function can be reduced to one line by removing the if()...end.

i.e., 
`(self.changed & ["periods_start", "periods_end", "periods_day", "periods_type"]).any?`

Comment (Bad-Science) [inline app/models/section.rb]: Use a % literal for string array here (see https://en.wikibooks.org/wiki/Ruby_Programming/Syntax/Literals#The_.25_Notation)

File: app/models/section.rb [hunk 2]
```diff
@@ -20,4 +21,16 @@ def conflicts_with(section)
     end
     false
   end
+
+  def sort_periods
+    periods_info = periods_day.zip(periods_start, periods_end, periods_type)
```
Comment (Bad-Science) [inline app/models/section.rb]: omit optional paranthesis

**Comment by Bad-Science** [comment]
@YushYush Additionally, please add unit tests for this. Unit tests for Section should be placed in a new file /spec/models/section_spec.rb and should verify that the sorting occurs when and how it is intended.

File: app/models/section.rb (hunk 2 above)
Comment (YushYush) [inline app/models/section.rb]: removed!

File: app/models/section.rb (hunk 1 above)
Comment (YushYush) [inline app/models/section.rb]: updates made!

File: app/models/section.rb [hunk 3]
```diff
@@ -20,4 +21,14 @@ def conflicts_with(section)
...
     false
   end
+
+  def sort_periods
+    periods_info = periods_day.zip periods_start, periods_end, periods_type
+    periods_info = periods_info.sort!.transpose
+    self.periods_day, self.periods_start, self.periods_end, self.periods_type = periods_info
+  end
+
+  def period_info_changed?
```
Comment (Bad-Science) [inline app/models/section.rb]: rename to `periods_changed?`

Comment (Bad-Science) [inline app/models/section.rb]: Pls change this method name so we can merge this in :)

**Comment by Bad-Science** [comment]
LGTM

**Pull request merged**